services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/api/containers.py

```diff
@@ -450,4 +450,65 @@ async def push_output_ports(
     return Response(status_code=status.HTTP_204_NO_CONTENT)
 
 
+@containers_router.post(
+    "/containers:restart",
+    response_model=None,
+    status_code=status.HTTP_204_NO_CONTENT,
+    responses={
+        status.HTTP_404_NOT_FOUND: {"description": "Container does not exist"},
+        status.HTTP_422_UNPROCESSABLE_ENTITY: {
+            "description": "Error while running docker-compose command"
+        },
+    },
+)
+async def restarts_containers(
+    command_timeout: float = Query(
+        10.0, description="docker-compose stop command timeout default"
+    ),
+    settings: DynamicSidecarSettings = Depends(get_settings),
+    shared_store: SharedStore = Depends(get_shared_store),
+    rabbitmq: RabbitMQ = Depends(get_rabbitmq),
+) -> Response:
+    """Removes the previously started service
+    and returns the docker-compose output"""
+
+    stored_compose_content = shared_store.compose_spec
+    if stored_compose_content is None:
+        raise HTTPException(
+            status.HTTP_404_NOT_FOUND,
+            detail="No spec for docker-compose command was found",
+        )
+
+    async def _run_command(command: str, command_timeout: Optional[float]) -> None:
+        finished_without_errors, stdout = await write_file_and_run_command(
+            settings=settings,
+            file_content=stored_compose_content,
+            command=command,
+            command_timeout=command_timeout,
+        )
+        if not finished_without_errors:
+            error_message = (f"'{command}' finished with errors\n{stdout}",)
+            logger.warning(error_message)
+            raise HTTPException(status.HTTP_422_UNPROCESSABLE_ENTITY, detail=stdout)
+
+    await _run_command(
+        command=(
+            "docker-compose --project-name {project} --file {file_path} "
+            "stop --timeout {stop_and_remove_timeout}"
+        ),
```

Comment:
you are aware that docker-compose has a command:
```
docker-compose restart --timeout 10 # defaults to 10
```
no need to call 2 functions.

Reply:
yes just realised it, also @pcrespov suggested it